### PR TITLE
chore(ci): add script for checking for unreleased changes on release branches

### PR DESCRIPTION
**Pull request by brettlangdon**
## Checklist

- [x] Change(s) are motivated and described in the PR description
- [x] Testing strategy is described if automated tests are not included in the PR
- [x] Risks are described (performance impact, potential for breakage, maintainability)
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed or label `changelog/no-changelog` is set
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/))
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))
- [x] If this PR changes the public interface, I've notified `@DataDog/apm-tees`.

## Reviewer Checklist

- [x] Title is accurate
- [x] All changes are related to the pull request's stated goal
- [x] Description motivates each change
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- [x] Testing strategy adequately addresses listed risks
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] Release note makes sense to a user of the library
- [x] Author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: scripts/unreleased-changes [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    changes = []
+
+    unreleased_version = None
+    for line in results.splitlines():
+        if line.startswith("v") and "-" in line:
+            unreleased_version = line.strip()
+        elif line.startswith("\t") and unreleased_version:
+            changes.append(line.strip())
+        elif unreleased_version and line.startswith("v"):
```
Comment (erikayasuda) [inline scripts/unreleased-changes:44]: NB: What case is this covering?

**Review by erikayasuda — APPROVED** [review]
One NB question

File: scripts/unreleased-changes [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+def get_unreleased_changes(branch: str) -> list[str]:
+    # get list of reno changes for the branch
+    results = subprocess.check_output(
+        ["reno", "--quiet", "list", "--branch", branch, "--stop-at-branch-base"], stderr=subprocess.PIPE
+    ).decode("utf-8")
+
+    changes = []
+
+    unreleased_version = None
+    for line in results.splitlines():
```
Comment (brettlangdon) [inline scripts/unreleased-changes:39]:
```suggestion
    # The output from reno list will be:
    #
    # v1.2.3-13
    #     releasenotes/unreleased-change
    # v1.2.3
    #     releasenotes/released-change ...
    # v1.2.2
    #     releasenotes/released-change ...
    #
    # We want to collect all the lines between the first version with a `vx.y.z-#` until
    # the next version
    for line in results.splitlines():
```

Comment (brettlangdon) [reply scripts/unreleased-changes:39]: @erikayasuda does this help?

File: scripts/unreleased-changes (hunk 1 above)
Comment (brettlangdon) [reply scripts/unreleased-changes:44]: added a code comment above, maybe that helps?

File: scripts/unreleased-changes (hunk 2 above)
Comment (erikayasuda) [reply scripts/unreleased-changes:39]: Yes! I tried running the command on my own and I didn't see this variation for some reason (probably didn't try enough versions)

**Pull request merged**